mirage/utcp_mirage.ml

```diff
@@ -57,36 +57,36 @@ module Make (Ip : Tcpip.Ip.S with type ipaddr = Ipaddr.t) = struct
 
   let read (t, flow) =
     match Utcp.recv t.tcp (now ()) flow with
-    | Ok (tcp, data, cond, segs) ->
+    | Ok (tcp, [], cond, segs) -> (
```

Comment:
See [this comment](https://github.com/robur-coop/utcp/pull/54/files#r2215330617). We are sure that all `Cstruct.t` given by `Utcp.recv` are **not** empty (see `Rope.to_css`). I noticed the case where `data` is empty before. Now we can take the advantage of the pattern-matching to split cases via a pattern instead of a `if Cstruct.length data == 0 ...`.